### ISSUE TITLE
Weird heading id generated for MD heading with curly braces

### ISSUE THREAD

**Issue by fflaten** [withastro/astro#5781]
### What version of `astro` are you using?

1.9.0

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Using curly braces in a heading in a `.md` file generates an function-like heading-id.

Example:

```mdx
# Some weird heading like { }

<!-- becomes -->
<h1 id="{$$slug(&#x60;Some" weird="" heading="" like="" ${="" }&#x60;)}="">Some weird heading like { }</h1>
```

Repro: Build and check `/dist/markdown/index.html`

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-cbd34s

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
The code that injected the strange `$$slug` string was removed as part of #5785, so this will be fixed in the next 2.0 beta release. Not sure if we need to backport anything to v1 or if it’s OK to say, upgrade to v2 to get the fix.

**Comment by delucis** [comment]
Cool, I think I agree, so will close as fixed. Thanks again for raising this and #5780 🙌
